src/components/post-card/PostCard.js

```diff
@@ -1,64 +1,78 @@
-import React from 'react';
+import React, {Component} from 'react';
 
-import { Comment } from '../comment/Comment';
+import {Comment} from '../comment/Comment';
 import DefaultImg from '../../assets/default-empty-img.png';
 import './PostCard.scss';
 
-// todo: переписать эту компоненту, чтоб она стала классовой
+// done_todo: переписать эту компоненту, чтоб она стала классовой
 //  в ней должно появиться свойство state
 //  в котором будет флажок showComments
 //  и будет кнопка, лейба которой будет либо "show comments" либо "hide comments"
 
-export function PostCard(props) {
-  const { post, hasImage, comments = [], author } = props;
+export class PostCard extends Component {
+    state = {
+        showComments: true,
+    }
 
-  console.log('hasImage', hasImage);
-  const { title, body } = post;
+    onShowHideClick = () => {
+        this.setState({
+            showComments: !this.state.showComments
+        })
+    };
 
-  const kittyUrl = `https://cataas.com/cat/says/hello%20world!?${Math.random() * 1000}`;
+    render() {
+        const {showComments} = this.state;
+        const {post, hasImage, comments = [], author} = this.props;
+        const {title, body} = post;
+        const kittyUrl = `https://cataas.com/cat/says/hello%20world!?${Math.random() * 1000}`;
+        // const renderImage = () => {
+        //     return hasImage ? <img src={kittyUrl} alt="cat"/> : <img src={DefaultImg} alt="DefaultImg"/>;
+        // };
 
-  const renderImage = () => {
-    return hasImage ? (
-      <img src={kittyUrl} />
-    ) : (
-      <img src={DefaultImg} />
-    );
-  };
+        return (
+            <div className="may-post-card card">
 
-  return (
-    <div className="may-post-card card">
-      {hasImage && (
-        <div className="may-post-card-img" id="my-block" onClick={() => {
-          alert('ghvcdhfvbdfsjvbdf');
-        }}>
-          <img src={kittyUrl} />
-        </div>
-      )
-      }
-      {!hasImage && (
-        <div className="may-post-card-img">
-          <img src={DefaultImg} />
-        </div>
-      )}
+                <button
+                    onClick={this.onShowHideClick}
+                    type="button"
+                    className="btn btn-outline-secondary ">
+                    hide/show comment
```

Comment:
Here should be label either `show comment` when showComments = false or `hide comments` when showComments = true

Reply:
You can use ternary operator  for that purpose

Reply:
Did it using the ternary operator.